### PR TITLE
Fix/time utils error

### PR DESCRIPTION
**Pull request by roesslerj**
Hopefully the least complex solution to the problem at hand ... to be discussed ;-)

### CODE REVIEW

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java [hunk 1]
```diff
@@ -22,14 +22,24 @@
   private static final Logger logger = LoggerFactory.getLogger(ProdPpacIosRateLimitStrategy.class);
   static final int VALIDITY_IN_HOURS = 23;
 
+  private final Clock clock;
+
+  public ProdPpacIosRateLimitStrategy() {
+    this.clock = Clock.systemUTC();
+  }
+
+  public ProdPpacIosRateLimitStrategy(Clock clock) {
```
Comment (FelixRottler) [inline services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java:33]: Maybe we need a Configuration class that injects for prod (based on a profile or a name) the no-arg constructor and for tests it injects the clock constructor. What do you think?

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java [hunk 2]
```diff
@@ -22,14 +23,25 @@
   private static final Logger logger = LoggerFactory.getLogger(ProdPpacIosRateLimitStrategy.class);
   static final int VALIDITY_IN_HOURS = 23;
 
+  private final Clock clock;
+
+  @Autowired
```
Comment (FelixRottler) [inline services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java:28]: Any particular reason for the use of Autowire here? I always thought one can use Autowire to perform Field Injection instead of constructor injection. My opinion is that the usage of Autowire should not be placed here.

Comment (roesslerj) [reply services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java:28]: Marks a constructor [...] as to be autowired by Spring's dependency injection facilities: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Autowired.html
This is also according to some code examples I found on the net...

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java (hunk 1 above)
Comment (roesslerj) [reply services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java:33]: We can do that if you want – but I see no benefits over this approach but added complexity.

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java (hunk 2 above)
Comment (FelixRottler) [reply services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java:28]: Ok I understand. then other change I suggested is not necessary

Comment (FelixRottler) [reply services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java:28]: How does this then work from inside the tests?

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java (hunk 1 above)
Comment (FelixRottler) [reply services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java:33]: Agree - based on the other comment

File: services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java (hunk 2 above)
Comment (roesslerj) [reply services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/ios/verification/scenario/ratelimit/ProdPpacIosRateLimitStrategy.java:28]: Yes. So this is the default for everything. Right now, we only need the "special" constructor to test a specific edge-case. So it only is used for that single test.

**Review by FelixRottler — APPROVED** [review]
